src/components/fabSpeedDial/fabSpeedDial.js

```diff
@@ -0,0 +1,130 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('material.components.fabSpeedDial', [])
+    .directive('mdFabSpeedDial', MdFabSpeedDialDirective);
+
+  /**
+   * @ngdoc directive
+   * @name mdFabSpeedDial
+   * @module material.components.fabSpeedDial
+   *
+   * @restrict E
+   *
+   * @description
+   * The `<md-fab-speed-dial>` directive is used present a series of popup elements (usually `<md-button>`s) for
+   * quick access to common actions.
+   *
+   * @usage
+   * <hljs lang="html">
+   * <md-fab-speed-dial direction="up">
+   *   <md-fab-trigger>
+   *     <md-button><md-icon icon="/img/icons/plus.svg"></md-icon></md-button>
+   *   </md-fab-trigger>
+   *
+   *   <md-fab-actions>
+   *     <md-button><md-icon icon="/img/icons/user.svg"></md-icon></md-button>
+   *     <md-button><md-icon icon="/img/icons/group.svg"></md-icon></md-button>
+   *   </md-fab-actions>
+   * </md-fab-speed-dial>
+   * </hljs>
+   *
+   * @param {string=} md-direction From which direction you would like the speed dial to appear relative to the
+   * trigger element.
+   * @param {expression=} md-open Programmatically control whether or not the speed-dial is visible.
+   */
+  function MdFabSpeedDialDirective() {
+
+    return {
+      restrict: 'E',
+
+      scope: {
+        direction: '@?mdDirection',
+        isOpen: '=?mdOpen'
+      },
+
+      controller: FabSpeedDialController,
+
+      compile: function($element) {
+        // Wrap every child in a new div and add a class that we can scale independently
+        var children = $element.find('md-fab-actions').children(),
+          child, i;
+
+        for (i = 0; i < children.length; i++) {
+          child = children.eq(i);
+
+          child.wrap('<div class="md-fab-action-item">');
+        }
+
+        // Return our link function
+        return {
+          post: link
+        }
+      }
+    };
+
+    function FabSpeedDialController($scope, $element) {
+      var vm = this;
+
+      vm.scope = $scope;
+      vm.element = $element;
+
+      // Set the default direction to 'down' if none is specified
+      if (typeof $scope.direction === 'undefined') {
+        $scope.$apply('direction = "down"');
```

Comment:
`$apply` is indeed necessary to notify Angular when something is invoked _from outside_.
Without getting into much detail, native/jqLite/jQuery event listener callbacks do need the `$apply`.
(Just make sure you are not calling the same functions from somewhere else while a digest cycle is already in progress.)

But they should be eliminated whenever possible (e.g. inside the ifs), because `$apply` does a great lot of stuff, which isn't necessary at that time.